widgets/common/src/DynamicDropdown.jsx

```diff
@@ -1,14 +1,44 @@
 // NOTE: Disabling react/require-default-props as default values are provided in component's definition
 /* eslint-disable react/require-default-props */
+import { useCallback } from 'react';
+import { debounce, isFunction } from 'lodash';
 import VisibilitySensor from 'react-visibility-sensor';
 import './DynamicDropdown.css';
 
 let instanceCount = 0;
+const defaultFetchState = { hasMore: true, currentPage: -1, shouldLoadMore: false };
+
+/**
+ * Creates two `useUpdateEffect` hooks to call `fetchTrigger` function with debouncing.
+ * First hook calls `fetchTrigger` with `reset` argument set to false on `withoutResetFetchDeps` change.
+ * The second calls `fetchTrigger` with `reset` argument set to true on `withResetFetchDeps` change
+ *
+ * @param {function(reset: boolean)} fetchTrigger function to be called to trigger data fetching
+ * @param {React.DependencyList} withoutResetFetchDeps list of dependencies for delayed fetch without reset
+ * @param {React.DependencyList} withResetFetchDeps list of dependencies for delayed fetch with reset
+ */
+function useFetchTrigger(fetchTrigger, withoutResetFetchDeps, withResetFetchDeps) {
+    const { useUpdateEffect } = Stage.Hooks;
+    const delayMs = 500;
+    const delayedFetchTrigger = useCallback(
+        debounce(reset => fetchTrigger(reset), delayMs),
```

Comment:
Optional:

Boolean parameters are confusing to me and IMO make the API harder to read :slightly_smiling_face: Just by reading the code, we have to trust that `false` means `reset`.

I suggest refactoring it to an object, which has a boolean property. This way there is an explicit contract and less knowledge required when using the hook

```suggestion
        debounce(reset => fetchTrigger({ reset }), delayMs),
```

or

```suggestion
        debounce(reset => fetchTrigger({ shouldReset : reset }), delayMs),
```

Reply:
I don't see a point in having an object here, in that case I would stick with single boolean argument. 
I will change it to `shouldReset` if that reads better.

Reply:
Maybe I didn't describe it well enough. For me, the argument is having an explicit API contract and fewer things to manually validate. If you have a boolean, who knows what is its meaning. You have to go to the function definition and check.

```
const fetchTrigger = (dontReset) => {...};

useFetchTrigger(fetchTrigger, ...);
```

It compiles, and reads well. Booleans match. But the expected behavior is exactly opposite :slightly_smiling_face: An API that uses objects reduces the overhead and the need for manual validation, because the object that is passed around is `{ shouldReset: boolean }`, and it's obvious what the meaning of `true` for that parameter is.

Anyway, I'll make the comment optional, change at your with :slightly_smiling_face:

Reply:
Changed.